x-pack/test/fleet_api_integration/apis/epm/template.ts

```diff
@@ -25,27 +25,36 @@ export default function ({ getService }: FtrProviderContext) {
     it('can be loaded', async () => {
       const template = getTemplate({
         type: 'logs',
-        templateName,
+        templateIndexPattern,
         mappings,
         packageName: 'system',
         composedOfTemplates: [],
+        templatePriority: 200,
       });
 
       // This test is not an API integration test with Kibana
       // We want to test here if the template is valid and for this we need a running ES instance.
       // If the ES instance takes the template, we assume it is a valid template.
-      const { body: response1 } = await es.indices.putTemplate({
-        name: templateName,
+      const { body: response1 } = await es.transport.request({
```

Comment:
why are the two ES requests here changed to the generic `transport.request`?

Reply:
We're working with index templates v2. The endpoint for these is `_index_template` instead of `_template`, and the new ES client provides `putIndexTemplate()` and `getIndexTemplate()` methods to interface with that. The integration tests use the legacy client, which doesn't have these methods, so I fall back to the generic method.

At some point someone(TM) should probably tackle getting the new ES client into the integration tests, I didn't because it didn't look obvious at first glance.

For index templates v2 see https://github.com/elastic/elasticsearch/issues/53101

Reply:
got it, thanks for that context. #74111 is the issue for migrating to new ES client, I just added a note about looking at usages in tests too